lib/spack/docs/packaging_guide.rst

```diff
@@ -2695,6 +2781,43 @@ the build will fail and the install prefix will be removed.  If they
 succeed, Spack considers the build successful and keeps the prefix in
 place.
 
+^^^^^^^^^^^^^^^^
+Build-time tests
```

Comment:
@tgamblin I'll put a note, but I think that the interface and API for *build-time* tests should not change drastically. Right now:
- a lot of things come for free
- running code after a phase requires just to use a decorator